test/fail_compilation/test16228.d

```diff
@@ -0,0 +1,24 @@
+/* REQUIRED_ARGS: -dip25
+   TEST_OUTPUT:
+---
+fail_compilation/test16228.d(22): Error: function has 'return' but does not return any indirections
+fail_compilation/test16228.d(23): Error: function test16228.S.foo non-static member has no 'this' to which 'return' can apply
```

Comment:
Another variant: `Error: function test16228.S.foo has 'return' but cannot return any instance data by reference, because it is static. 'return' is only applicable to non-static member functions`.